chainstate/src/detail/mod.rs

```diff
@@ -83,6 +86,7 @@ pub struct Chainstate<S, V> {
     custom_orphan_error_hook: Option<Arc<OrphanErrorHandler>>,
     events_controller: EventsController<ChainstateEvent>,
     time_getter: TimeGetter,
+    is_initial_block_download_finished: AtomicBool,
```

Comment:
I don't think this needs to be atomic.

Reply:
It is needed for "interior mutability", so this flag can be changed in the non-mut function. Do you think it would be better to change the function signature to `is_initial_block_download(&mut self)`?

Reply:
I am not sure whether we should go for `&mut self` or not but if we go down the interior mutability route, I'd prefer it to be `Cell<bool>`. It communicates the intent better. An atomic type kind of suggests you access the field from multiple threads.

Reply:
No, we shouldn't. We don't have to copy bitcoin. We can just update this value after every new block.